geode-redis/src/main/java/org/apache/geode/redis/internal/netty/NettyRedisServer.java

```diff
@@ -0,0 +1,317 @@
+    }
+    workerGroup.shutdownGracefully();
+    Future<?> bossFuture = bossGroup.shutdownGracefully();
+    if (serverChannel != null) {
+      serverChannel.close();
+    }
+    bossFuture.syncUninterruptibly();
+    if (closeFuture != null) {
+      closeFuture.syncUninterruptibly();
+    }
+  }
+
+  public int getPort() {
+    return serverPort;
+  }
+
+  private ChannelInitializer<SocketChannel> createChannelInitializer() {
+    String redisPassword = configSupplier.get().getRedisPassword();
+    final byte[] redisPasswordBytes = Coder.stringToBytes(redisPassword);
+
+    return new ChannelInitializer<SocketChannel>() {
+      @Override
+      public void initChannel(SocketChannel socketChannel) {
+        if (logger.isDebugEnabled()) {
+          logger.debug(
+              "GeodeRedisServer-Connection established with " + socketChannel.remoteAddress());
+        }
+        ChannelPipeline pipeline = socketChannel.pipeline();
+        addSSLIfEnabled(socketChannel, pipeline);
+        pipeline.addLast(ByteToCommandDecoder.class.getSimpleName(), new ByteToCommandDecoder());
+        pipeline.addLast(new WriteTimeoutHandler(10));
+        pipeline.addLast(ExecutionHandlerContext.class.getSimpleName(),
+            new ExecutionHandlerContext(socketChannel, regionProvider, pubsub, subscriberGroup,
+                allowUnsupportedSupplier, shutdownInvoker, redisPasswordBytes));
+      }
+    };
+  }
+
+  private void addSSLIfEnabled(SocketChannel ch, ChannelPipeline p) {
+
+    SSLConfig sslConfigForComponent =
+        SSLConfigurationFactory.getSSLConfigForComponent(configSupplier.get(),
+            SecurableCommunicationChannel.SERVER);
+
+    if (!sslConfigForComponent.isEnabled()) {
+      return;
+    }
+
+    SslContext sslContext;
+    try {
+      KeyStore ks = KeyStore.getInstance("JKS");
+      ks.load(new FileInputStream(sslConfigForComponent.getKeystore()),
+          sslConfigForComponent.getKeystorePassword().toCharArray()/**/);
+
+      // Set up key manager factory to use our key store
+      KeyManagerFactory kmf =
+          KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+      kmf.init(ks, sslConfigForComponent.getKeystorePassword().toCharArray());
+
+      SslContextBuilder sslContextBuilder = SslContextBuilder.forServer(kmf);
+      sslContext = sslContextBuilder.build();
+
+    } catch (KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException | IOException
+        | CertificateException e) {
+      throw new RuntimeException(e);
+    }
+    p.addLast(sslContext.newHandler(ch.alloc()));
+  }
+
+  private Class<? extends ServerChannel> initializeEventLoopGroups() {
+    ThreadFactory selectorThreadFactory =
+        new LoggingThreadFactory("GeodeRedisServer-SelectorThread-", false);
+
+    ThreadFactory workerThreadFactory =
+        new LoggingThreadFactory("GeodeRedisServer-WorkerThread-", true);
+
+    if (singleThreadPerConnection) {
```

Comment:
What do you think about removing this altogether? We do no testing with this mode and I wouldn't expect anyone to actually use it.

Reply:
I like that idea. It uses deprecated apis so I'd really want us to test with it before allowing customers to use it.
But I think that should be on its own PR. We have one about numThreads sys prop; seems like we could do it on that one.

Reply:
I added this to the numThreads sys prop story

Reply:
Sounds good.